src/main/frontend/components/ReviewTile.js

```diff
@@ -12,13 +19,80 @@ const ReviewTile = (props) => {
     userReview = "User Review: "
   }
 
+  const voted = count => {
+    let localCount = stateCount
+    if (downvoted == true && count == 1 && upvoted == false) {
+      setDownvoted(false)
+      setUpvoted(true)
+      localCount += 2
+    } else if (upvoted == true && count == -1 && downvoted == false) {
+      setUpvoted(false)
+      setDownvoted(true)
+      localCount -= 2
+    } else if (count == 1 && upvoted == true) {
+      setUpvoted(false)
+      setDownvoted(false)
+      localCount -= 1
+    } else if (count == 1 && upvoted == false) {
+      setUpvoted(true)
+      setDownvoted(false)
+      localCount += 1
+    } else if (count == -1 && downvoted == true) {
+      setDownvoted(false)
+      setUpvoted(false)
+      localCount += 1
+    } else if (count == -1 && downvoted == false) {
+      setDownvoted(true)
+      setUpvoted(false)
+      localCount -= 1
+    }
+    setCount(localCount)
+    updateVoteCount(id, localCount)
+  }
+
+  const updateVoteCount = async (id, finalCount) => {
+    try {
+      const response = await fetch(`/api/v1/reviews/${id}`, {
+        method: 'PUT',
+        headers: new Headers({
+          'Content-Type': 'application/json'
+        }),
+        body: JSON.stringify({ id: id, finalCount: finalCount })
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+        } else {
+          const errorMessage = `${response.status} (${response.statusText})`
+          const error = new Error(errorMessage)
+          throw (error)
+        }
+      }

```

Comment:
After the `updatedVoteCount` returns a 200 response, we can then update the `stateCount` with the correct count that we know has been persisted.